meta/src/main/proto/gravitino_meta.proto

```diff
@@ -61,10 +61,35 @@ message Schema {
   uint64 id = 1;
   string name = 2;
   AuditInfo audit_info = 3;
+  // These two fields will be used when the schema object is managed by Gravitino, otherwise, it
+  // will be ignored.
+  optional string comment = 4;
+  map<string, string> properties = 5;
 }
 
 message Table {
   uint64 id = 1;
   string name = 2;
   AuditInfo audit_info = 3;
-}
\ No newline at end of file
+}
+
+/**
+ * The Fileset message is used to record the fileset object information. Fileset object represents
+ * the non-tabular data on FS-like storage, like HDFS, S3, etc.
+ */
+message Fileset {
+  enum Type {
+    MANAGED = 0; // The storage location is managed by Gravitino. When specified, the data will be
+                 // deleted when the file object is deleted.
+    EXTERNAL = 1; // The storage location is managed by user. When specified, the data will not be
+                  // deleted when the file object is deleted.
+  }
+
+  uint64 id = 1;
+  string name = 2;
+  optional string comment = 3;
+  Type type = 4;
+  optional string storage_location = 5;
```

Comment:
Should we provide a default storage_location if storage_location is absent?

Reply:
We will use the catalog/schema's location to figure out a location for fileset if it is not set.